src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php

```diff
@@ -170,27 +170,27 @@ public function getRequestQuery()
 
     public function getRequestHeaders()
     {
-        return new HeaderBag($this->data['request_headers']);
+        return new ParameterBag($this->data['request_headers']);
```

Comment:
As decided in previous PRs about the profiler internals, we don't want to preserve bc here, that would slow us too much.